### PR TITLE
quick-start and user-guide sections with placeholders

### PR DESCRIPTION
**Pull request by jkbradley**
Included some code snippets, including graph construction and BFS.
Also added constructor from edges only, plus test.
Minor user guide cleanups.

### CODE REVIEW

**Comment by jkbradley** [comment]
CC @thunterdb   I would not bother running the code snippets in the docs yet.  Let's check those once the GraphX API wrappers are cleaned up since the APIs may change a little bit.

I'd mainly check the new GraphFrame constructor from edges, its test, and the basic layout of the Markdown files.  Thanks!

File: docs/index.md [hunk 1]
```diff
@@ -11,6 +11,19 @@ It aims to provide both the functionality of GraphX and extended functionality t
...
 serialization, and highly expressive graph queries.
 
+# What are GraphFrames?
+
+*GraphX is to RDDs as GraphFrames are to DataFrames.*
+
+GraphFrames represent graphs: vertices (e.g., users) and edges (e.g., relationships between users).
+If you are familiar with [GraphX](http://spark.apache.org/docs/latest/graphx-programming-guide.html),
+then GraphFrames will be easy to learn.  The key difference is that GraphFrames are based upon
+[Spark DataFrames](http://spark.apache.org/docs/latest/sql-programming-guide.html), rather than RDDs.
```
Comment (thunterdb) [inline docs/index.md]: maybe link to RDDs as well, for users that are new and unfamiliar?

File: docs/user-guide.md [hunk 2]
```diff
@@ -0,0 +1,443 @@
...
+layout: global
+displayTitle: GraphFrames User Guide
+title: User Guide
+description: GraphFrames GRAPHFRAMES_VERSION user guide
+---
+
+This page gives examples of how to use GraphFrames for basic queries, motif finding, and
+general graph algorithms.  This includes code examples in Scala and Python.
+
+* Table of contents
```
Comment (thunterdb) [inline docs/user-guide.md]: You mean `##`?

File: docs/user-guide.md [hunk 3]
```diff
@@ -0,0 +1,443 @@
...
+{:toc}
+
+*Note: Most examples use the GraphFrame from the first subsection:
+[Creating GraphFrames](user-guide.html#creating-graphframes).*
+
+# Creating GraphFrames
+
+Users can create GraphFrames from vertex and edge DataFrames.
+* *Vertex DataFrame*: A vertex DataFrame should contain a special column named "id" which specifies
```
Comment (thunterdb) [inline docs/user-guide.md]: You should use `-` instead of `*`, the list is not rendering

File: docs/user-guide.md [hunk 4]
```diff
@@ -0,0 +1,443 @@
...
+  unique IDs for each vertex in the graph.
+* *Edge DataFrame*: An edge DataFrame should contain two special columns: "src" (source vertex ID
+  of edge) and "dst" (destination vertex ID of edge).
+
+Both DataFrames can have arbitrary other columns.  Those columns can represent vertex and edge
+attributes.
+
+A GraphFrame can also be constructed from edge DataFrame, without a vertex DataFrame specified.
```
Comment (thunterdb) [inline docs/user-guide.md]: ... from a single DataFrame containing edge information. The vertices will be inferred from the source and destinations of the edges.

File: docs/user-guide.md [hunk 5]
```diff
@@ -0,0 +1,443 @@
...
+<div class="codetabs">
+
+The following example demonstrates how to create a GraphFrame from vertex and edge DataFrames.
+
+<div data-lang="scala"  markdown="1">
+{% highlight scala %}
+// Vertex DataFrame
+val v = sqlContext.createDataFrame(List(
+  ("a", "Alice", 34),
+  ("b", "Bob", 36),
```
Comment (thunterdb) [inline docs/user-guide.md]: `Joseph` is missing :-)

**Comment by thunterdb** [comment]
I just have small style comments.

Regarding the examples, I suggest we add a `friends` dataframe in the package, so that we can write fully functional examples instead of ellipsis:

```scala
import org.graphframes.examples
val friends: GraphFrames = examples.friends

// Get a DataFrame with columns "id" and "inDeg" (in-degree)
val vertexInDegrees: DataFrame = g.inDegrees

...
```

**Comment by thunterdb** [comment]
The last comment is for the next PRs of course.

**Comment by thunterdb** [comment]
Feel free to merge the PR after that.

File: docs/user-guide.md (hunk 2 above)
Comment (jkbradley) [inline docs/user-guide.md]: No, it's just text to be scraped. I'll note that.

File: docs/user-guide.md (hunk 3 above)
Comment (jkbradley) [inline docs/user-guide.md]: Fixed (and checked generated doc).  I forgot to have a newline before the list.

**Comment by jkbradley** [comment]
Thanks!  I'll merge it now.  I'll try to add the friends thing in the next PR I send.

**Pull request merged**